### PR TITLE
Port SWE4DS exercises

### PR DESCRIPTION
**Pull request by astrojuanlu**
For now just

- Copy pasted instructions and code
- Ported tests from unittest to pytest
- Minor modifications to top-level README

This needs a second pass to verify the ergonomics on Gitpod or our platform of choice.

See https://github.com/kedro-org/kedro-devrel/issues/35.

### CODE REVIEW

File: swe4ds/configuration/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Use configuration
+
+You were given some code that reads a bunch of files and applies some basic preprocessing.
```
Comment (stichbury) [inline swe4ds/configuration/README.md:3]:
```suggestion
You were given some code that reads a set of files and applies some basic preprocessing.
```

File: swe4ds/environment-dependencies/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Environment and dependencies
+
+Your client is trying to run a very old code delivered by a former colleague, but they're unable to do so.
```
Comment (stichbury) [inline swe4ds/environment-dependencies/README.md:3]:
```suggestion
Your client is trying to run very old code delivered by a former colleague, but they're unable to get it working.
```

File: swe4ds/environment-dependencies/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+3. Now all the dependencies should be in place, but there's an error with the line `from sklearn.gaussian_process import GaussianProcess`. The reason is that the model name changed after this code was written. We could change the line to `from sklearn.gaussian_process import GaussianProcessRegressor`, but more code changes would be needed because of changes introduced in newer scikit-learn versions, and it's not allowed. Take a deep breath.
+4. Remember that this code is old: therefore, an older version of scikit-learn will do the trick! Install `scikit-learn` 0.19 and rerun the script.
+
+Finally! That was a bit frustrating but at least an `errorbars.png` appears in the filesystem, and the script finishes (although with lots of deprecation warnings).
+
+### Declare the dependencies
+
+Create a `requirements.txt` file with the dependencies you have to install. To verify that it works:
```
Comment (stichbury) [inline swe4ds/environment-dependencies/README.md:32]: Do you want to say something about where to store it (like, "in the same folder as the notebook" or "within a known location"? I know we're not assuming anything about their folder structure/project at this point, so I'm not sure, but I feel like we should offer some guidance.

**Review by stichbury — APPROVED** [review]

File: swe4ds/environment-dependencies/README.md (hunk 3 above)
Comment (astrojuanlu) [reply swe4ds/environment-dependencies/README.md:32]: Indeed, will add some clarification here

Comment (astrojuanlu) [reply swe4ds/environment-dependencies/README.md]: Done!

**Pull request merged**